### ISSUE TITLE
[DynamicPage]: [A11y][SF] title is exposed as a button when it’s not interactive

### ISSUE THREAD

**Issue by sleric1024** [SAP/ui5-webcomponents#12253]
### Describe the bug

When only a title is provided to DynamicPage (no subtitle/subHeader etc.), the title element is not clickable, yet it’s rendered with role="button" (and focusable). Screen readers announce it as a “button,” which is misleading and fails accessibility conformance.

### Isolated Example

https://stackblitz.com/edit/github-crdamynp?file=src%2FApp.tsx

### Reproduction steps

Go to: https://stackblitz.com/edit/github-crdamynp?file=src%2FApp.tsx
Using Chrome Dev Tool to inspect the title area.

### Screenshots or Videos

<img width="913" height="696" alt="Image" src="https://github.com/user-attachments/assets/9cfc6cb1-1ba0-4216-bf82-08284dfa98e5" />

### UI5 Web Components for React Version

2.13.2

### UI5 Web Components Version

2.13.3

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.

**Comment by duygu-rmdn** [comment]
Hi @UI5/ui5-team-ctr-pir, could you please take a look?

**Comment by NakataCode** [comment]
Hello @sleric1024,

This issue has been fixed in PR #12141 and is available in UI5 Web Components version [2.14.0](https://github.com/UI5/webcomponents/releases/tag/v2.14.0).

Best regards,
Nikola
